sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -199,4 +199,6 @@ case class RDDScanExec(
   override def simpleString: String = {
     s"$nodeName${truncatedString(output, "[", ",", "]")}"
   }
+
+  override def inputRDD: RDD[InternalRow] = rdd
```

Comment:
The test failure is real.  `RDDScanExec.rdd` may not be RDD of unsafe row. Maybe we should enforce that, `InputRDDCodegen.inputRDD` is `RDD[UnsafeRow]`.